R/functions_util.R

```diff
@@ -58,28 +59,70 @@ scrnaseq_session_info = function(path_to_git=".") {
   return(out)
 }
 
+#' Returns a subsample of cells.
+#' 
+#' @param sc A Seurat sc object.
+#' @param num Number of cells to subsample.
+#' @param seed Seed for sampling. Default is 1.
+#' @param group Metadata colum group to consider for sampling with group_proportional.
+#' @param group_proportional Should the number of cells sampled from each group be proportional (TRUE) or should the number of cells be the same for each group (FALSE)? Only works if group is not NULL.
+#' @return Sampled cell names.
+ScSampleCells = function(sc, n, seed=1, group=NULL, group_proportional=TRUE, group_min_cells=0) {
```

Comment:
My idea was that especially for heatmaps we sample similar numbers of cells per group so that visualisation is better and big cluster do not take up so much space.  I would keep it as function so that in future we can do more complex sampling.

Reply:
Ok, I understand and agree. Still I find it good to always try and keep it simple as the code gets bigger and bigger.